src/tools/build_helper/src/ci.rs

```diff
@@ -19,6 +19,15 @@ impl CiEnv {
     pub fn is_ci() -> bool {
         Self::current() != CiEnv::None
     }
+
+    /// Checks if running in upstream-managed CI job.
```

Comment:
Not sure what we mean by "upstream" here. Maybe we can just say "rust-lang/rust official CI", or something like that?